samples/Kconfig

```diff
@@ -13,6 +13,7 @@ config NCS_SAMPLES_DEFAULTS
 	imply ASSERT_NO_COND_INFO
 	imply ASSERT_NO_MSG_INFO
 	imply HW_STACK_PROTECTION if ARCH_HAS_STACK_PROTECTION
+	imply FPU if SPM
```

Comment:
@oyvindronningstad (and people in general) Do you think this is a sane imply? Or should it be more linked to `CONFIG_SPM_SERVICE_RNG` which is the config that `select FPU`

Reply:
I think we should raise a build error when the FPU configs mismatch. It's a bit more involved, but the error is so hard to debug, and we have no sure way to prevent it in just kconfig.

Reply:
With the build error in place I think it's better to place it with the Rng config.

Reply:
So if `CONFIG_SPM_SERVICE_RNG` does `select FPU if IS_SPM`, then perhaps it should also `select/imply FPU if SPM`? I guess our customer applications are not likely to use `NCS_SAMPLES_DEFAULTS`.

Reply:
```
config SPM_SERVICE_RNG
	bool "Request random numbers"
	default y if SPM
	select NORDIC_SECURITY_BACKEND if IS_SPM
	select FPU if IS_SPM
```
The thing is that SPM's configuration is separate from the application's so we can't have this entry `select` something for the application, we'd have to have an entry for SPM, e.g.:

```
config SPM
	bool "Use Secure Partition Manager"
	default y if TRUSTED_EXECUTION_NONSECURE
	depends on !BUILD_WITH_TFM
	select FW_INFO
	imply ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS if SPM_SECURE_SERVICES
	select FPU <--
```

Perhaps we could set `default n` for `SPM_SERVICE_RNG` and let those samples that need it provide a "child configuration" for SPM and manually set `CONFIG_FPU=y` in their `prj.conf`. Perhaps that's better than adding the FPU requirement to everything.

Reply:
just removed IS_SPM from the service which selects FPU in SPM so that FPU is set in both.